core/src/query/store.rs

```diff
@@ -288,3 +286,47 @@ impl LiveQueryStoreHandle {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::num::NonZeroU32;
+
+    use super::*;
+
+    #[test]
+    fn query_message_order_preserved() {
+        let query_store = LiveQueryStore::test();
+        let threaded_rt = tokio::runtime::Runtime::new().unwrap();
+        let query_store_handle = threaded_rt.block_on(async { query_store.start() });
+
+        for i in 0..10_000 {
+            let pagination = Pagination::default();
+            let fetch_size = FetchSize {
+                fetch_size: NonZeroU32::new(1),
+            };
+            let sorting = Sorting::default();
+
+            let query_output = LazyValue::Iter(Box::new((0..100).map(|_| Value::Bool(false))));
```

Comment:
Why do you need `Box` here?

Reply:
Because of `Box<dyn Iterator>` type in `LazyValue::Iter`